packages/terra-date-picker/src/react-datepicker/day.jsx

```diff
@@ -16,25 +16,70 @@ import {
 
 class Day extends React.Component {
   static propTypes = {
+    /**
+     * Date Value Entered.
+     */
     day: PropTypes.object.isRequired,
+    /**
+     * A callback function for custom day class names.
+     */
     dayClassName: PropTypes.func,
+    /**
+     * Maximum Date value for a date range.
+     */
     endDate: PropTypes.object,
+    /**
+     * Highlight range of dates with custom classes.
+     */
     highlightDates: PropTypes.instanceOf(Map),
+    /**
+     * Prop to render Inline version of date picker component.
+     */
     inline: PropTypes.bool,
     /**
      * @private
      * Internationalization object with translation APIs. Provided by `injectIntl`.
      */
     intl: intlShape,
+    /**
+     * Month value for the date entered.
+     */
     month: PropTypes.number,
+    /**
+     * A callback function to execute when a valid date is selected.
```

Comment:
This should be "A callback function to execute when mouse is clicked to select date."